domain/blockdevice/state/state.go

```diff
@@ -32,20 +34,94 @@ func NewState(factory coredatabase.TxnRunnerFactory) *State {
 	}
 }
 
-// BlockDevices returns the BlockDevices for the specified machine.
+// GetBlockDevice retrieves the info for the specified block device.
+//
+// The following errors may be returned:
+// - [blockdeviceerrors.BlockDeviceNotFound] when the block device is not found.
+func (st *State) GetBlockDevice(
+	ctx context.Context, uuid blockdevice.BlockDeviceUUID,
+) (coreblockdevice.BlockDevice, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return coreblockdevice.BlockDevice{}, errors.Capture(err)
+	}
+
+	input := entityUUID{
+		UUID: uuid.String(),
+	}
+	blockDeviceStmt, err := st.Prepare(`
+SELECT &blockDevice.*
+FROM   block_device
+WHERE  uuid = $entityUUID.uuid
+`, input, blockDevice{})
+	if err != nil {
+		return coreblockdevice.BlockDevice{}, errors.Capture(err)
+	}
+
+	blockDeviceLinkStmt, err := st.Prepare(`
+SELECT &deviceLink.*
+FROM   block_device_link_device
+WHERE  block_device_uuid = $entityUUID.uuid
+`, input, deviceLink{})
+	if err != nil {
+		return coreblockdevice.BlockDevice{}, errors.Capture(err)
+	}
+
+	var blockDevice blockDevice
```

Comment:
Nit: our convention is to get these above the prepare and use them in the prepare. I am fine with this landing without it. Just an FYI.